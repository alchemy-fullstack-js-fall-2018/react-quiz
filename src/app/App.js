import React from 'react';
import Shape from '../shape/Shape';

export default function app() {
  return (
    <Shape />
  );
}
