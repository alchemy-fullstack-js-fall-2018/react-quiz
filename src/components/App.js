import React, { Component, Fragment } from 'react';
import ColorSelector from './ColorSelector';
import TextSelector from './TextSelector';
import Shape from './Shape';

export default class App extends Component {
  state = {
    fontColor: '',
    backgroundColor: '',
    text: ''
  };

  onChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  };

  render() {
    const { fontColor, backgroundColor, text } = this.state;

    return (
      <Fragment>
        <ColorSelector
          fontColor={fontColor}
          backgroundColor={backgroundColor}
          onChange={this.onChange}
        />
        <TextSelector text={text} onChange={this.onChange} />
        <Shape
          fontColor={fontColor}
          backgroundColor={backgroundColor}
          text={text}
        />
      </Fragment>
    );
  }
}
