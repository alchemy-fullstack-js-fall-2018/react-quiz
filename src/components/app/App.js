import React, { Component } from 'react';

import TextSelector from '../textSelector/TextSelector.js';
import ColorSelector from '../colorSelector/ColorSelector.js';
import Shape from '../shape/Shape.js';

export default class App extends Component {

  state = {
    text: '',
    fontColor: 'black',
    backgroundColor: 'white'
  }


  onChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  };

  render() {

    const { text, fontColor, backgroundColor } = this.state;

    return (
      <main>
        <TextSelector
          text={text}
          onChange={this.onChange}
        /> <br/>
        <ColorSelector
          fontColor={fontColor}
          backgroundColor={backgroundColor}
          onChange={this.onChange}
        /> <br />
        <Shape
          text={text}
          color={fontColor}
          backgroundColor={backgroundColor}
        />
      </main>
    );
  }


}
