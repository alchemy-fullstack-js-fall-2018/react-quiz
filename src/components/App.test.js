import React from 'react';
import App from './App';
import render from 'react-test-renderer';

describe('App Component Test', () => {
  test('App snapshot', () => {
    expect(render
      .create(<App />)
      .toJSON()).toMatchSnapshot();
  });
});
