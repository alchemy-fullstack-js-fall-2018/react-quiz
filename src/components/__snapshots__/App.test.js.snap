// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Test App snapshot 1`] = `
Array [
  <div>
    <label
      name="fontColor"
    >
      Choose a font color:
    </label>
    <input
      name="fontColor"
      onChange={[Function]}
      type="color"
      value="white"
    />
    <label
      name="backgroundColor"
    >
      Choose a background color:
    </label>
    <input
      name="backgroundColor"
      onChange={[Function]}
      type="color"
      value="grey"
    />
  </div>,
  <div>
    <label
      name="text"
    >
      Text:
    </label>
    <input
      name="text"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>,
  <div
    style={
      Object {
        "background": "grey",
        "color": "white",
      }
    }
  >
    
  </div>,
]
`;

exports[`Shape Component Test Shape snapshot 1`] = `
<div
  style={
    Object {
      "background": undefined,
      "color": undefined,
    }
  }
/>
`;
